pkg/handler/storage/client.go

```diff
@@ -0,0 +1,437 @@
+		return nil, errors.OAuth2InvalidRequest("filestorage is being deleted")
+	}
+
+	organizationID := current.Labels[coreconstants.OrganizationLabel]
+	projectID := current.Labels[coreconstants.ProjectLabel]
+	regionID := current.Labels[constants.RegionLabel]
+
+	required, err := c.generateV2(ctx, organizationID, projectID, regionID, request)
+	if err != nil {
+		return nil, err
+	}
+
+	updated := current.DeepCopy()
+	updated.Labels = required.Labels
+	updated.Annotations = required.Annotations
+	updated.Spec = required.Spec
+
+	s := newUpdateSaga(c, organizationID, regionID, current, updated)
+	if err := saga.Run(ctx, s); err != nil {
+		return nil, err
+	}
+
+	return convertV2(updated), nil
+}
+
+func (c *Client) Delete(ctx context.Context, storageID string) error {
+	resource, err := c.GetRaw(ctx, storageID)
+	if err != nil {
+		return err
+	}
+
+	organizationID := resource.Labels[coreconstants.OrganizationLabel]
+	projectID := resource.Labels[coreconstants.ProjectLabel]
+
+	if err := rbac.AllowProjectScope(ctx, "region:filestorage:v2", identityapi.Delete, organizationID, projectID); err != nil {
+		return err
+	}
+
+	if resource.DeletionTimestamp != nil {
+		return nil
+	}
+
+	if err := c.client.Delete(ctx, resource); err != nil {
+		return errors.OAuth2ServerError("delete failed").WithError(err)
+	}
+
+	return nil
+}
+
+func (c *Client) generateAllocation(size int) identityapi.ResourceAllocationList {
+	return identityapi.ResourceAllocationList{
+		{
+			Kind:      "filestorage",
+			Committed: size,
+		},
+	}
+}
+
+func (s *createSaga) createAllocation(ctx context.Context) error {
+	required := s.client.generateAllocation(s.request.Spec.Size)
+
+	return identityclient.NewAllocations(s.client.client, s.client.identity).Create(ctx, s.filestorage, required)
+}
+
+func (s *createSaga) deleteAllocation(ctx context.Context) error {
+	if err := identityclient.NewAllocations(s.client.client, s.client.identity).Delete(ctx, s.filestorage); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *createSaga) createFileStorage(ctx context.Context) error {
+	if err := s.client.client.Create(ctx, s.filestorage); err != nil {
+		return errors.OAuth2ServerError("unable to create filestorage").WithError(err)
+	}
+
+	return nil
+}
+
+func (s *createSaga) validateRequest(ctx context.Context) error {
+	storageClassKey := client.ObjectKey{
+		Name:      s.request.Spec.StorageClassId,
+		Namespace: s.client.namespace,
+	}
+	storageType := regionv1.FileStorageClass{}
+	networkClient := network.New(s.client.client, s.client.namespace, s.client.identity)
+
+	err := s.client.client.Get(ctx, storageClassKey, &storageType)
+
+	if err != nil {
+		return errors.OAuth2ServerError("filestorage class does not exist").WithError(err)
+	}
+
+	for _, v := range s.request.Spec.Attachments.NetworkIDs {
+		_, err = networkClient.GetV2(ctx, v)
+		if err != nil {
+			return errors.OAuth2ServerError("network attachment ID not found").WithError(err)
```

Comment:
```suggestion
			return errors.OAuth2ServerError("network attachment not found").WithError(err).WithLabels("networkID", v)
```

The WithErr and WithLabels values get logged, but not included in the API response.